util/tracer_nvbit/tracer_tool/tracer_tool.cu

```diff
@@ -321,26 +328,52 @@ void nvbit_at_cuda_event(CUcontext ctx, int is_exit, nvbit_api_cuda_t cbid,
         active_region = false;
     }
 
-    if (user_defined_folders == 1) {
-      std::string usr_folder = std::getenv("TRACES_FOLDER");
-      std::string temp_traces_location = usr_folder;
-      std::string temp_kernelslist_location = usr_folder + "/kernelslist";
-      std::string temp_stats_location = usr_folder + "/stats.csv";
-      traces_location.resize(temp_traces_location.size());
-      kernelslist_location.resize(temp_kernelslist_location.size());
-      stats_location.resize(temp_stats_location.size());
-      traces_location.replace(traces_location.begin(), traces_location.end(),
-                              temp_traces_location);
-      kernelslist_location.replace(kernelslist_location.begin(),
-                                   kernelslist_location.end(),
-                                   temp_kernelslist_location);
-      stats_location.replace(stats_location.begin(), stats_location.end(),
-                             temp_stats_location);
-      printf("\n Traces location is %s \n", traces_location.c_str());
-      printf("Kernelslist location is %s \n", kernelslist_location.c_str());
-      printf("Stats location is %s \n", stats_location.c_str());
+    if (multi_process){
+        std::string command = "ls -d " + trace_folder + "/run*/ 2>/dev/null | wc -l";
+        FILE* pipe = popen(command.c_str(), "r");
+        if (!pipe) {
+            std::cerr << "Error: Failed to execute command.\n";
+            return ;
+        }
+    
+        int dir_count = 0;
+        fscanf(pipe, "%d", &dir_count); // Read the count
+        pclose(pipe);
+        trace_folder += "/run" + std::to_string(dir_count);
+      
+      
+      if (mkdir(trace_folder.c_str(), S_IRWXU | S_IRWXG | S_IROTH | S_IXOTH) == -1) {
+        if (errno == EEXIST) {
+          // alredy exists
+        } else {
+          // something else
+          std::cout << "cannot create folder error:" << strerror(errno)
+                    << std::endl;
+          return;
+        }
+      }
+
     }
 
+
+    std::string temp_traces_location = trace_folder;
```

Comment:
Why do we need all this post-processing?
Can't we just set the trace location properly once and avoid all this re-work?